imports/plugins/included/payments-stripe/client/checkout/components/injectedCardForm.js

```diff
@@ -0,0 +1,273 @@
+      invalid_cvc: i18next.t("checkout.errorMessages.invalidCVC", "Your card's CVC is invalid."),
+      invalid_expiry_month: i18next.t("checkout.errorMessages.invalidExpiryMonth", "Your card's expiration month is invalid."),
+      invalid_expiry_year: i18next.t("checkout.errorMessages.invalidExpiryYear", "Your card's expiration year is invalid."),
+      invalid_expiry_month_past: i18next.t("checkout.errorMessages.incompleteExpiryMonthPast", "Your card's expiration date is in the past."),
+      invalid_expiry_year_past: i18next.t("checkout.errorMessages.incompleteExpiryYearPast", "Your card's expiration year is in the past."),
+      invalid_number: i18next.t("checkout.errorMessages.invalidNumber", "Your card number is invalid."),
+      postal_code_invalid: i18next.t("checkout.errorMessages.postalCodeInvalid", "Your card's postal code is invalid."),
+      state_unsupported: i18next.t("checkout.errorMessages.stateUnsupported", "Your state is not supported."),
+      whoops: i18next.t("checkout.errorMessages.whoops", "Whoops! Something went wrong. Please try again.")
+    };
+  }
+
+  handleSubmit = (ev) => {
+    const resubmitMessage = "Resubmit payment";
+    ev.preventDefault();
+
+    if (this.state.cardNumberErrorMessage || this.state.expDateErrorMessage || this.state.CVVErrorMessage || this.state.postalErrorMessage) {
+      return;
+    }
+
+    this.setState({
+      submitMessage: "Submitting...",
+      submitting: true
+    });
+    if (this.props.stripe) {
+      this.props.stripe.createToken().then((payload) => {
+        if (payload.error) {
+          this.setState({
+            errorMessage: "",
+            submitMessage: resubmitMessage,
+            submitting: false
+          });
+          // Do not duplicate field validation errors
+          if (payload.error.type !== "validation_error") {
+            this.setState({ errorMessage: payload.error.message });
+          }
+          return;
+        }
+        Meteor.apply("stripe/payment/createCharges", ["authorize", payload.token, this.props.cartId], {
+          wait: true,
+          onResultReceived: (error, result) => {
+            if (error || (result && result.error)) {
+              this.setState({
+                errorMessage: this.errorCodes[result.error.code] ? this.errorCodes[result.error.code] : result.error.whoops,
+                submitMessage: resubmitMessage,
+                submitting: false
+              });
+            } else {
+              Router.go("cart/completed", {}, {
+                _id: this.props.cartId
+              });
+            }
+          }
+        });
+        return;
+      }).catch(() => {
+        this.setState({
+          errorMessage: "Something went wrong. Please try again.",
+          submitMessage: resubmitMessage,
+          submitting: false
+        });
+        return;
+      });
+    }
+    return;
+  };
+
+  handleCardNumberChange = (change) => {
+    if (change && change.error) {
```

Comment:
this `if` block can be extracted to a function and params passed in. And then use on all the change handlers to reduce code duplication.